### PR TITLE
Updated stylesheets+preference packs.

### PR DESCRIPTION
**Pull request by MisterMakerNL**
Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [X]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---
I updated the theme files to a new standard.
Also added a lot of new/missing icons.
It's a lot of code so it probably has some bugs. But I am sure it has less issues then the once we use now.
I do have some issues with the new styles and preferences getting picked up.

Screenshot of the 5 styles:
afbeelding
afbeelding
afbeelding
afbeelding
afbeelding

### CODE REVIEW

**Review by MisterMakerNL — COMMENTED** [review]
Looks good now!

**Comment by adrianinsaval** [comment]
would png files not be problematic when scaling? Would it be hard to make those as svg too?

File: src/Gui/PreferencePacks/package.xml [hunk 1]
```diff
@@ -3,18 +3,32 @@
   <name>Built-In Preference Packs</name>
   <description>Preference Packs included with the FreeCAD distribution</description>
   <version>1.0.0</version>
-  <maintainer email="email@freecad.org">No Maintainer</maintainer>
+  <maintainer email="email@freecad.org">MisterMaker</maintainer>
```
[the remark this answers is not in the document]
Comment (MisterMakerNL) [reply src/Gui/PreferencePacks/package.xml:6]: I don't really like to put my mail address in public spots. Spam bots and such will pick it up. Contact can go true the forum or github.

[the remark this answers is not in the document]
Comment (adrianinsaval) [reply src/Gui/PreferencePacks/package.xml:6]: Understandable

**Comment by MisterMakerNL** [comment]
> 

Good point and I had a look at it and came across some junk files that shouldn't be included. 
Will fix it tomorrow.

**Comment by MisterMakerNL** [comment]
> would png files not be problematic when scaling? Would it be hard to make those as svg too?

update!

**Comment by chennes** [comment]
I called it "FreeCAD Classic Colors" to make sure it was clear that Preference Pack modified just color information. I forsee a time when there are lots of kinds of Packs out there, configuring many different things about FreeCAD. So I'm very hesitant to scale that name all the way back to just "Classic". Classic what? Keyboard shortcuts? Toolbar layouts?

**Comment by MisterMakerNL** [comment]
> 

Ugh I keep forgetting we can use other stuff with preference packs! We really need an example of a non-theme preference pack.
Would calling theme theme's be something?
- Classic theme
- Dark theme
- Light theme

we can go with:
- Classic colors
- Dark colors
- Light colors
But it sounds kinda cheap.

**Comment by MisterMakerNL** [comment]
> 

The classic removes stylesheets and sets it to none, the others apply a stylesheet although removing stylsesheets and apply is the same, since no-stylesheet is also a stylesheet although it is the default.
So I feel like it changes more than colors. 
Also theme's are in my opinion more common used description for such features. So in my opinion it is not just about what it technically is, but how a user would expect how it is called. 
Most users are known with theme's in browsers and on their phones, so calling it theme's is more logical.

**Comment by MisterMakerNL** [comment]
I updated it, renamed it to Classic theme and Dark/light theme also adjusted the description and tags a bit.

**Comment by MisterMakerNL** [comment]
Removed trailing white spaces.

File: src/Gui/PreferencePacks/package.xml [hunk 2]
```diff
@@ -3,17 +3,39 @@
...
+      <version>0.5.0</version>
+      <tag>built-in</tag>
+      <tag>dark</tag>
+      <tag>background</tag>
+      <tag>stylesheet</tag>
+      <tag>theme</tag>
+    </preferencepack>
+    <preferencepack>
+      <name>Light theme</name>
+      <description>Applies a light background and a darker stylesheets, you can choose other light stylesheets as you wish.</description>
```
Comment (chennes) [inline src/Gui/PreferencePacks/package.xml:32]: darker = lighter?

**Comment by MisterMakerNL** [comment]
I had some issues while testing realized I didn't rename the cfg files, now everything seems to work.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9715">pipeline status</a> for feature branch [PR_9715](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_9715). Pipeline [903566385](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/903566385) was triggered at [00b00a4](https://github.com/FreeCAD/FreeCAD/pull/9715/commits/00b00a40c01a498bc76ff569fd87a9ba2d5a2871). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by luzpaz** [comment]
@chennes is this slated to be merged in v0.22 ?

**Comment by chennes** [comment]
https://github.com/orgs/FreeCAD/projects/4

**Comment by marioalexis84** [comment]
The light, dark, darker and modern styles have the calltips in the Python console too transparent.
calltip

**Comment by MisterMakerNL** [comment]
That was a pretty annoying too fix, the idea of using transparant was so that I got a very uniform darkness over all the theme's.
This basically broke that. Now dark light and darker has their own darkness setting. Still kept in the transparency since it felt kinda fancy and more modern. I'll add this to my other pr.
afbeelding
afbeelding
afbeelding

**Comment by marioalexis84** [comment]
In some systems, with the light, dark, darker and modern styles, I see these static vertical and horizontal lines.
Even being system dependent, since the lines don't exist with the others stylesheets, they could be fixed.
black_lines

**Comment by MisterMakerNL** [comment]
Could you elaborate on which systems this happens?

**Comment by adrianinsaval** [comment]
https://forum.freecad.org/viewtopic.php?t=79716
I think we should consider reverting this PR for the time being and do it properly early next cycle

**Comment by MisterMakerNL** [comment]
I'm open to improve on it having so much feedback is useful, something that you don't get when it's not in main.
But that some hardcore users have issues with adjusting to change and now are nitpicking it, like they are just minor visual bugs normal users won't even notice them. The height of the buttons are 2px different. Lets pull the pr.

I'm sure it never is going to be perfect but I'm sure this pr with some minor fixes is way better than original. If you undo the pr with such limited review time and try to make it perfect you not just demotivate the hell out of me, you just made a stand that it's your Freecad and this process is more about pleasing some core users instead of making things better.

**Comment by adrianinsaval** [comment]
> I'm sure it never is going to be perfect but I'm sure this pr with some minor fixes is way better than original. If you undo the pr with such limited review time and try to make it perfect you not just demotivate the hell out of me, you just made a stand that it's your Freecad and this process is more about pleasing some core users instead of making things better.

if you look at my response in that thread I pointed out a lot more issues than the arrow, I'm willing to adapt to the new dropdown arrow (but prefer the old style) but there's several other things that make it look noticeable worse for me. I listed all I could find so that it is actionable and not just some vague sentiment like "I prefer the old one". Since there's many details that need ironing out it is my personal opinion (just my opinion) that it would be better to do this immediately after this release so we don't have to rush it now before release.

I understand that having your work reverted can be very discouraging and I truly do not want to discourage you, I only propose reverting for a little time until release, not a definitive rejection of this. I have to say that for me it was unexpected that this PR got merged, we were supposed to be in feature freeze already. This PR did not fix any bug that I'm aware of, and it's not a trivial one that can be just merged without thought. The only thing I see is this very vague statement:

> It's a lot of code so it probably has some bugs. But I am sure it has less issues then the once we use now.

with which I have to disagree after testing.

@sliptonic any comments on that?

**Comment by MisterMakerNL** [comment]
This kins of feedback is really useful to perfect the stylesheets.
But all those tiny things don't break any functionality, and can be fixed pretty fast without impacting any other code.
Your screenshots show that you are using out of date stylesheets.
I already modified the tabs on your request.

Some discussions about it:
Triangle pull downs icon. Yeah or no?
Icons spacing vs screen space.

**Comment by sliptonic** [comment]
> https://forum.freecad.org/viewtopic.php?t=79716 I think we should consider reverting this PR for the time being and do it properly early next cycle

Based on what?   The PR was (mostly) created properly.  It sat for almost a month without comment.  It passed all unit tests.  Now people are unhappy and want to revert it based on value judgments.  That's not how the development process works.

I agree with many of the comments raised here and in the forum but I disagree with reverting.  Pull Requests have consequences and people should take the time to test things.

@MisterMakerNL  I said that the PR was 'mostly' created properly.  There are two notable exceptions.   I can't find an issue where the problems you're trying to fix are addressed.   Second, the PR is larger than needed.  Many of these changes could have been spread out over numerous issues/PRs and implemented incrementally.  You would have gotten better and more timely feedback that way.

**Comment by adrianinsaval** [comment]
> I already modified the tabs on your request.

What I showed on the pictures is what got merged, I can only assume that you forgot to push those changes to the PR branch. Besides that, you did modify the tabs to be more intuitive but personally I still think they look like buttons and they shouldn't. This is of course my own personal take, I do not claim authority over what tabs should look like.

> But all those tiny things don't break any functionality, and can be fixed pretty fast without impacting any other code.

It does not hinder functionality but it seriously affects the experience, IMO there's too little time for thorough testing of something that can have effects everywhere in the interface, we're only scratching the surface of the software and already found many things that make the experience less pleasant. The previous themes where working fine, perhaps they aren't as maintainable long term, but such refactoring should be done early in a development cycle not at the last stage.

> Based on what? The PR was (mostly) created properly. It sat for almost a month without comment. It passed all unit tests. Now people are unhappy and want to revert it based on value judgments. That's not how the development process works.

It's not a value judgement, I think what the PR aims to do is right, but as it is it brings more problems than it solves so it is not succeeding in what it set out to do. I still don't understand why the PR was merged, I did not even try to review this because my assumption was that we where in feature freeze and this PR would be for after release, IIRC the freeze had started in may. This is not a bugfix.

> Pull Requests have consequences and people should take the time to test things.

I agree, but this was not the time for this. We are wrapping things up for release, it is not the time for a significant change that requires significant testing. The existing themes had fixes for many hidden niche things that were reported over the years, it'll take some time for the refactored theme to get up to par.

**Comment by sliptonic** [comment]
I merged the PR.  It was done last week during a cleanup of the release project backlog.  It might have been a mistake because we were moving fast.  

The value judgements I'm talking about aren't here but in the forum where there was also discussion of reversion.  IMHO, all the following are examples.   They all express valid perspectives but would do nothing more than frustrate me if I was the author and got them after a month of waiting for feedback on the PR.


"...these chevrons have appeared for the toolbar drop down lists as shown in this example.  I find them very clumsy and make it so that I need to be way more careful ..."

"...Are these chevrons supposed to make it appear more modern or just make it harder to use? ...or both? ..."

"...probably a different style that will resemble an arrow a less a "visual glitch" will be more visually appealing...."

"...This does not improve usability for me. It actually impedes my work. I should not have to study where the mouse is once it has entered the border of the icon. ..."

"...but besides that there is a loss in the clarity of where you need to click to trigger what action,..."

"...About the modern theme, please let's keep tabs looking like actual tabs :? this is an stablished useful GUI concept, why go against it? ..."

"...the headers in the property view look huge in comparison to the more important tabs in the task dialog, I suggest making these headers significantly smaller, ..."

"...I've said this before but disabled icons look very strange like this, I don't think they should have a different background...."

**Pull request merged**